### ISSUE TITLE
Warning: Accessing non-existent property 'MongoError' of module exports inside circular dependency

### ISSUE THREAD

**Issue by OliverwengFiltered** [Automattic/mongoose#9900]
Upgraded to mongoose 5.11.15, and getting this warnning.
Warning: Accessing non-existent property 'MongoError' of module exports inside circular dependency.

Anyone has insights about this one ?

**Comment by Megapixel99** [comment]
I am running into this error too.

**Comment by xkmgt** [comment]
A temporarily fix for this is https://stackoverflow.com/a/66102270/10944219

**Comment by vkarpov15** [comment]
This is an issue with the MongoDB driver: https://github.com/mongodb/node-mongodb-native/pull/2734 . We'll upgrade to MongoDB 3.6.5 when it is released :+1:

**Comment by Akshit-WTF** [comment]
Upgrading to version Mongoose@5.11.16 has brought this issue to my workspace. However, downgrading back to 5.11.15 fixes it. Also, 5.11.16 brings along a new depreciation warning:
`DeprecationWarning: Listening to events on the Db class has been deprecated and will be removed in the next major version.`

**Comment by ddialar** [comment]
I can confirm the situation exposed by @Akshit-WTF. Same update and sale warning message.

I followed the downgrading process proposed in the previous message and the warning disappeared.

**Comment by MichaelGrigoryan25** [comment]
I was getting the same error but after downgrading to v5.11.13 it started working

**Comment by EmanSza** [comment]
Warning Still Exists on 5.11.16. Downgrading to 5.11.15 Fixed Warning.
![image](https://user-images.githubusercontent.com/44507685/108771814-25e87680-752a-11eb-86fd-c006bc487a27.png)

**Comment by MichaelGrigoryan25** [comment]
Yeah I still get the warning and after tracing the issue here's the result

```
node --trace-warnings src/index
(node:14900) Warning: Accessing non-existent property 'MongoError' of module exports inside circular dependency
    at emitCircularRequireWarning (internal/modules/cjs/loader.js:650:11)
    at Object.get (internal/modules/cjs/loader.js:664:5)
    at Object.<anonymous> (C:\Users\micha\Desktop\????\node_modules\mongodb\lib\operations\operation.js:4:38)      
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (C:\Users\micha\Desktop\????\node_modules\mongodb\lib\operations\execute_operation.js:4:16)
```

**Comment by EmanSza** [comment]
> Yeah I still get the warning and after tracing the issue here's the result
> 
> ```
> node --trace-warnings src/index
> (node:14900) Warning: Accessing non-existent property 'MongoError' of module exports inside circular dependency
>     at emitCircularRequireWarning (internal/modules/cjs/loader.js:650:11)
>     at Object.get (internal/modules/cjs/loader.js:664:5)
>     at Object.<anonymous> (C:\Users\micha\Desktop\????\node_modules\mongodb\lib\operations\operation.js:4:38)      
>     at Module._compile (internal/modules/cjs/loader.js:1063:30)
>     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
>     at Module.load (internal/modules/cjs/loader.js:928:32)
>     at Function.Module._load (internal/modules/cjs/loader.js:769:14)
>     at Module.require (internal/modules/cjs/loader.js:952:19)
>     at require (internal/modules/cjs/helpers.js:88:18)
>     at Object.<anonymous> (C:\Users\micha\Desktop\????\node_modules\mongodb\lib\operations\execute_operation.js:4:16)
> ```
Is the warning on .15 or .16?

**Comment by MichaelGrigoryan25** [comment]
> > Yeah I still get the warning and after tracing the issue here's the result
> > ```
> > node --trace-warnings src/index
> > (node:14900) Warning: Accessing non-existent property 'MongoError' of module exports inside circular dependency
> >     at emitCircularRequireWarning (internal/modules/cjs/loader.js:650:11)
> >     at Object.get (internal/modules/cjs/loader.js:664:5)
> >     at Object.<anonymous> (C:\Users\micha\Desktop\????\node_modules\mongodb\lib\operations\operation.js:4:38)      
> >     at Module._compile (internal/modules/cjs/loader.js:1063:30)
> >     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
> >     at Module.load (internal/modules/cjs/loader.js:928:32)
> >     at Function.Module._load (internal/modules/cjs/loader.js:769:14)
> >     at Module.require (internal/modules/cjs/loader.js:952:19)
> >     at require (internal/modules/cjs/helpers.js:88:18)
> >     at Object.<anonymous> (C:\Users\micha\Desktop\????\node_modules\mongodb\lib\operations\execute_operation.js:4:16)
> > ```
> 
> Is the warning on .15 or .16?

It's an internal issue in mongoose/mongodb drivers

`node_modules\mongodb\lib\operations\execute_operation.js:4:16`

**Comment by EmanSza** [comment]
> > > Yeah I still get the warning and after tracing the issue here's the result
> > > ```
> > > node --trace-warnings src/index
> > > (node:14900) Warning: Accessing non-existent property 'MongoError' of module exports inside circular dependency
> > >     at emitCircularRequireWarning (internal/modules/cjs/loader.js:650:11)
> > >     at Object.get (internal/modules/cjs/loader.js:664:5)
> > >     at Object.<anonymous> (C:\Users\micha\Desktop\????\node_modules\mongodb\lib\operations\operation.js:4:38)      
> > >     at Module._compile (internal/modules/cjs/loader.js:1063:30)
> > >     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
> > >     at Module.load (internal/modules/cjs/loader.js:928:32)
> > >     at Function.Module._load (internal/modules/cjs/loader.js:769:14)
> > >     at Module.require (internal/modules/cjs/loader.js:952:19)
> > >     at require (internal/modules/cjs/helpers.js:88:18)
> > >     at Object.<anonymous> (C:\Users\micha\Desktop\????\node_modules\mongodb\lib\operations\execute_operation.js:4:16)
> > > ```
> > 
> > 
> > Is the warning on .15 or .16?
> 
> It's an internal issue in mongoose/mongodb drivers
> 
> `node_modules\mongodb\lib\operations\execute_operation.js:4:16`

yes i know its a internal issue but downgarding to .15 should fix the warning

**Comment by MichaelGrigoryan25** [comment]
> > > > Yeah I still get the warning and after tracing the issue here's the result
> > > > ```
> > > > node --trace-warnings src/index
> > > > (node:14900) Warning: Accessing non-existent property 'MongoError' of module exports inside circular dependency
> > > >     at emitCircularRequireWarning (internal/modules/cjs/loader.js:650:11)
> > > >     at Object.get (internal/modules/cjs/loader.js:664:5)
> > > >     at Object.<anonymous> (C:\Users\micha\Desktop\????\node_modules\mongodb\lib\operations\operation.js:4:38)      
> > > >     at Module._compile (internal/modules/cjs/loader.js:1063:30)
> > > >     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
> > > >     at Module.load (internal/modules/cjs/loader.js:928:32)
> > > >     at Function.Module._load (internal/modules/cjs/loader.js:769:14)
> > > >     at Module.require (internal/modules/cjs/loader.js:952:19)
> > > >     at require (internal/modules/cjs/helpers.js:88:18)
> > > >     at Object.<anonymous> (C:\Users\micha\Desktop\????\node_modules\mongodb\lib\operations\execute_operation.js:4:16)
> > > > ```
> > > 
> > > 
> > > Is the warning on .15 or .16?
> > 
> > 
> > It's an internal issue in mongoose/mongodb drivers
> > `node_modules\mongodb\lib\operations\execute_operation.js:4:16`
> 
> yes i know its a internal issue but downgarding to .15 should fix the warning

Well it doesn't seem to work with version .15 either

**Comment by vkarpov15** [comment]
Fixed by #10038
